samples/pushBatchOfDocumentsFromFolder.ts

```diff
@@ -37,18 +33,13 @@ async function main() {
     '/path/to/fileB.json',
   ];
 
-  const callback: UploadBatchCallback = (
-    err: unknown,
-    data: UploadBatchCallbackData
-  ) => {
-    if (err) {
-      onErrorCallback(err, data);
-    } else {
-      onSuccessCallback(data);
-    }
-  };
-
-  await source.batchUpdateDocumentsFromFiles('my_source_id', entries, callback);
+  const {onBatchUpload, onBatchError, done} =
+    await source.batchUpdateDocumentsFromFiles('my_source_id', entries);
+
+  onBatchUpload(onSuccessCallback);
+  onBatchError(onErrorCallback);
+  await done();
```

Comment:
I think the builder pattern would be more 'fun' to use?
i.e. 
batchUpdateDocumentsFromFiles return a Promise-like object with two additional methods:
`onBatchUpload` and `onBatchError`, which return the same Promise-like object.

The code here would be become:
```suggestion
  await source.batchUpdateDocumentsFromFiles('my_source_id', entries)
    .onBatchUpload(onSuccessCallback)
    .onBatchError(onErrorCallback);
```